src/include/homestore/vchunk.h

```diff
@@ -0,0 +1,38 @@
+/*********************************************************************************
+ * Modifications Copyright 2017-2019 eBay Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <sisl/fds/buffer.hpp>
+#include <homestore/blk.h>
+#include <homestore/homestore_decl.hpp>
+
+namespace homestore {
+class Chunk;
+
+class VChunk {
+public:
+    VChunk(cshared< Chunk >&);
```

Comment:
We probably aren't going to mutate the pointer passed in to the caller on their behalf correct? So this should be either:
`cshared<Chunk> const&` or `cshared<Chunk>&&`
that way someone can do something like:
```
... VChunk(std::make_shared<Chunk>(...)) ... ;
```
requiring the pointer to be mutable means it needs to be a non-temporary on the callers stack (for no reason that I can see).

Reply:
alson, this should be `explicit` right?

Reply:
yes you are right, `const &` probabaly better than rvalue reference as it can still accept left value.

Reply:
> also, this should be `explicit` right?

it think it is better to keep this implicity for convenience , see https://github.com/eBay/HomeStore/blob/8065a05407c0715609a6405d36715d9c65a61c82/src/lib/device/virtual_dev.cpp#L102


other comments makes sense, fixed